bin/bl-exit

```diff
@@ -141,643 +43,126 @@ if display:
-
-            self.tooltips.set_tip(button, str(btype))
-            return icon_w
-
-        def get_image_info(self, img_path):
-            '''Test if icon is png, and get icon width(px),icon height(px)'''
-            width = -1
-            height = -1
-            try:
-                with open(img_path, 'rb') as f:
-                    data = f.read()
-                if data[:8] == '\211PNG\r\n\032\n' and (data[12:16] == 'IHDR'): # check if png
-                    width, height = struct.unpack('>LL', data[16:24])
-            except:
-                self.on_warning("get_image_info failed for '{}'".format(img_path))
-            return width,height
-
+# Translate command-line option to method - command line only
+actionToMethod = dict(
+	cancel='Cancel', c='Cancel',
+	logout='Logout', l='Logout',
+	suspend='Suspend', s='Suspend',
+	hybridsleep='HybridSleep', y='HybridSleep',
+	hibernate='Hibernate', i='Hibernate',
+	reboot='Reboot', b='Reboot',
+	poweroff='PowerOff', p='PowerOff'
+)
 
 def print_message(m):
-        print (str(m), file=sys.stderr)
+		print (str(m), file=sys.stderr)
 
 def get_options():
-    result = None
-    import argparse
-    parser = argparse.ArgumentParser(description="Bunsenlabs exit")
-    if display:
-        parser.add_argument("-l", "--logout", help="Log out",
-                            action="store_true")
-    parser.add_argument("-s", "--suspend", help="Suspend",
-                        action="store_true")
-    parser.add_argument("-i", "--hibernate", help="Hibernate",
-                        action="store_true")
-    parser.add_argument("-y", "--hybridsleep", help="Hybrid sleep",
-                        action="store_true")
-    parser.add_argument("-b", "--reboot", help="Reboot",
-                        action="store_true")
-    parser.add_argument("-p", "--poweroff", help="Power off",
-                        action="store_true")
-    parser.parse_args(sys.argv[1:])
-    """No check if more than one option was specified. Take the first option and
-    discard the other"""
-    result = parser.parse_args()
-    return result
+	result = None
+	parser = argparse.ArgumentParser(description="Bunsenlabs exit")
+	if display:
+		parser.add_argument("-l", "--logout", help="Log out",
+							action="store_true")
+		parser.add_argument("-s", "--suspend", help="Suspend",
+							action="store_true")
+		parser.add_argument("-i", "--hibernate", help="Hibernate",
+							action="store_true")
+		parser.add_argument("-y", "--hybridsleep", help="Hybrid sleep",
+							action="store_true")
+		parser.add_argument("-b", "--reboot", help="Reboot",
+							action="store_true")
+		parser.add_argument("-p", "--poweroff", help="Power off",
+							action="store_true")
+		parser.parse_args(sys.argv[1:])
+	"""No check if more than one option was specified. Take the first option and
+	discard the other"""
+	result = parser.parse_args()
+	return result
 
 def get_config_file():
```

Comment:
Let's rewrite this function to use python-xdg, it provides a more complete implementaiton and stable interface. Also, this function completely fails on my system and returns None (Arch Linux, latest Python 2.7).